src/mon/OSDMonitor.cc

```diff
@@ -7289,6 +7292,7 @@ int OSDMonitor::prepare_command_pool_application(const string &prefix,
 
   p.last_change = pending_inc.epoch;
   pending_inc.new_pools[pool] = p;
+  *modified = true;
```

Comment:
I don't hold a strong opinion on whether 'out' arguments should be pointers or references, but this one should either be a reference or protected by an `assert(modified != nullptr)`

Reply:
Our style guide says out params should be pointers, and all references should be const. :)